### ISSUE TITLE
In place update of a value in a multi-document yaml file nukes other documents

### ISSUE THREAD

**Issue by DeathRabbit679** [mikefarah/yq#847]
Summary:
Here's a simple example of what I've seen.
```
$ cat test.yml
---
foo:
  bar: 1
---
foo:
  bar: 2
---
foo:
  bar: 3
$ docker run --rm -v $(pwd):/workdir mikefarah/yq:latest eval -i  'select(documentIndex == 2) | .foo.bar = 10' test.yml
$ cat test.yml
foo:
  bar: 10
```
Perhaps there's another flag or operator I'm missing to get this to be have like I expected, but I would have expected the result to be 
```
---
foo:
  bar: 1
---
foo:
  bar: 2
---
foo:
  bar: 10
```

Version of yq: 
```
$ docker run --rm -v $(pwd):/workdir mikefarah/yq:latest --version
yq version 4.9.3
```

**Comment by DeathRabbit679** [comment]
As an additional note, using the `--doc 2` in v3, the file retains the other documents after being edited in place

**Comment by mikefarah** [comment]
You need to put brackets to make sure the document selection happens just on the LHS of the assignment - as opposed to filtering to only that document, and then piping to an assignment.

TLDR:

```
yq e '(select(documentIndex == 2) | .foo.bar) = 10'
```
